tests/python/test_ad_for.py

```diff
@@ -327,7 +327,7 @@ def double_for():
         assert b.grad[i] == total_grad_b
 
 
-@ti.require(ti.extension.adstack)
+@ti.require(ti.extension.adstack, ti.extension.data64)
```

Comment:
Just curious, why only this case needs the data64 extension?

Reply:
It's because of this `cast<f64>`. Unfortunately, Metal doesn't support `double` yet: https://github.com/taichi-dev/taichi/blob/df62174854af8f5d964b39f0239e0f909810110a/tests/python/test_ad_for.py#L351

Reply:
Thanks for the clarification!